### ISSUE TITLE
Rational' can't be assigned to a variable of type 'Decimal 

### ISSUE THREAD

**Issue by halenoor96** [onepub-dev/money.dart#70]
of type 'Rational' can't be assigned to a variable of type 'Decimal'.
 - 'Rational' is from 'package:rational/rational.dart' ('/C:/Users/Hale%20Noor/AppData/Local/Pub/Cache/hosted/pub.dartlang.org/rational-2.2.0/lib/rational.dart').
 - 'Decimal' is from 'package:decimal/decimal.dart' ('/C:/Users/Hale%20Noor/AppData/Local/Pub/Cache/hosted/pub.dartlang.org/decimal-2.3.0/lib/decimal.dart').
      (amount * Decimal.ten.pow(amount.scale)).toBigInt(),

I am using latest version of money2 package but I found this issue.
Kindly Resolve it as soon as possible.
Thankx

**Comment by bsutton** [comment]
Can you provide an example that democrats the problem.

**Comment by bgetsug** [comment]
The error thrown during compilation appears to be happening in the `fixed` package, if the `decimal` package was upgraded to 2.3.0:

```
    ../../flutter/.pub-cache/hosted/pub.dartlang.org/fixed-2.2.1/lib/src/fixed.dart:97:29: Error: A value of type 'Rational' can't be assigned to a variable of type 'Decimal'.
     - 'Rational' is from 'package:rational/rational.dart' ('../../flutter/.pub-cache/hosted/pub.dartlang.org/rational-2.2.0/lib/rational.dart').
     - 'Decimal' is from 'package:decimal/decimal.dart' ('../../flutter/.pub-cache/hosted/pub.dartlang.org/decimal-2.3.0/lib/decimal.dart').
          (amount * Decimal.ten.pow(amount.scale)).toBigInt(),
                                ^
```

My workaround is to update my `pubspec.yaml` to override the version of `decimal` to 2.2.0:
```yaml
dependency_overrides:
  decimal: 2.2.0 
```

**Comment by bsutton** [comment]
@bgetsug 
I'm still having trouble reproducing this problem.

What version of dart are you running?

Are you able to supply a small project that demonstrates the problem.

@a14n do you have any ideas about this one?

**Comment by bsutton** [comment]
@bgetsug 
I've now managed to reproduce the bug and it looks like a bug in the decimal package. I've reported the issue to the decimal maintainers.

**Comment by a14n** [comment]
This error should be fixed by https://github.com/onepub-dev/fixed/pull/9 . Meanwhile you can limit the version of decimal with `>=2.0.0 <2.3.0`'

**Comment by bsutton** [comment]
This has been fixed in money 3.3.0 and fixed 2.3.0

Thanks to @14n for providing the patch.
